src/interface/report/Results/DegradedExperience.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from 'interface/layout/Theme.scss';
+import Danger from 'interface/common/Alert/Danger';
+import MODULE_ERROR from 'parser/core/MODULE_ERROR';
+
+class DegradedExperience extends React.PureComponent {
+  static propTypes = {
+    disabledModules: PropTypes.object.isRequired,
+  };
+
+  constructor(...args){
+    super(...args);
+    this.state = {
+      expanded: false,
+    };
+    this.toggleDetails = this.toggleDetails.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps){
+    this.forceUpdate();
```

Comment:
This is both considered bad practice in React and unnecessary, since it receiving props will already trigger a rerender except if shouldComponentUpdate is implemented. As it happens, it is because at the top of this class your extending React.PureComponent rather than React.Component, which automatically does that for you. If you change that to React.Component, shouldComponentUpdate will be cleared and this will rerender every time it gets new props.

Reply:
oh nice I was struggling with getting it to update like that and couldn't think of an easy fix for that (relatively new to react), I'll implement that real quick!